Выполнил всё задание по спринту, реализовав всё, что требовалось

Ruslan051015 opened the pull request:
Реализовал конечный автомат пользовательского интерфейса, если конечно правильно использую определение, а если простым языком то, вроде как дела всё, что нужно было в данном спринте, дико сложно было, без подсказок бы не справился(

Ruslan051015 committed:
Выполнил всё задание по спринту, реализовав всё, что требовалось

mmbabaev commented on MovieQuiz/Presentation/MovieQuizViewController.swift:
```diff
@@ -1,72 +1,163 @@
 
 final class MovieQuizViewController: UIViewController {
-    // MARK: - Lifecycle
+//Аутлеты
```
<img width="870" alt="Screen Shot 2023-03-06 at 19 18 03" src="https://user-images.githubusercontent.com/12600807/223172004-f50ccb03-875c-4150-abd3-66538f4146b2.png">
принято помечать секции кода через MARK, также он красиво умеет отображаться в xcode

mmbabaev commented on MovieQuiz/Presentation/MovieQuizViewController.swift:
```diff
@@ -1,72 +1,163 @@
+    @IBOutlet private weak var imageView: UIImageView!
+    @IBOutlet private weak var textLabel: UILabel!
+    @IBOutlet private weak var counterLabel: UILabel!
+//Действия
```
порядок в viewController обычно делают такой: 

1. Properties
2. override funcs
3. IBAction funcs
4. other funcs  другие типы - сверху или снизу

mmbabaev commented on MovieQuiz/Presentation/MovieQuizViewController.swift:
```diff
@@ -1,72 +1,163 @@
+        textLabel.text = step.question
+        counterLabel.text = step.questionNumber
+   }
+// Прописываем реализацию метода показа результата квиза
```
предлагаю не писать такие обучающие комментарии

https://habr.com/ru/company/piter/blog/458990/

Ruslan051015 replied:
Понимаю смысл замечания, самому не особо они нравятся, просто пока что пишу их, для тренировки самого себя и чтобы не запутаться, в дальнейшем конечно же их не будет в коде

Ruslan051015 committed:
Исправил замечание касаемо скрепления рамки и картинки, теперь все скруглено.
Перетасовал порядок во "ViewController"
Пометил секции кода коментов "MARK: -"
Добавил отключение кнопок после выбора варианта ответа, правда возможно они как-то грубо висят в коде и их следовало вынести в отдельную функцию, но я честно не хотел намудрить лишнего, главное работает пока)
Комменты с пояснением все-таки оставил, пока мне с ними легче ориентироваться, в дальнейшем буду старится меньше писать их
